### PR TITLE
ci(build.yml): fix armv7 packages' architecture

### PR DESCRIPTION
**Pull request by MarksonHon**
### Checklist

- [x] The Pull Request has been fully tested
- [ ] There's an entry in the CHANGELOG
- [ ] There is a user-facing docs PR against https://github.com/daeuniverse/daed

### Full changelog

- [Implement ...]

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

**Comment by MarksonHon** [comment]
DON'T merge this pull request until fpm fix [pacman's build bug](https://github.com/jordansissel/fpm/pull/2017)

**Comment by kunish** [comment]
OK, noted, we can mark this PR as DRAFT for now to be extra safe

**Review by yqlbu — DISMISSED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -196,8 +196,13 @@ jobs:
             sudo apt install -y libarchive-tools rpm
             sudo gem install fpm -v 1.15.1
```
Comment (kunish) [inline .github/workflows/build.yml:197]: Note: dont forget to update fpm when the new release gets published

**Comment by MarksonHon** [comment]
Finished those jobs:
1. Fix armv7 packages' architectures, it should be `armhf` in Debian, `arm7hf` in Arch Linux ARM and `armhfp` in RPM.
2. Arch Linux ARM and RPM use `aarch64` for `arm64` architecture, while Debian use `arm64`.
3. Don't build MIPS family's pacman package for Arch Linux MIPS port has died for 10 years.
This PR can be merged now

**Comment by kunish** [comment]
@yqlbu instantiate-check-runs is causing PR build failure, please help

**Comment by yqlbu** [comment]
@kunish Sure, let me take a closer look

**Comment by yqlbu** [comment]
> @yqlbu instantiate-check-runs is causing PR build failure, please help

The root cause is that the `feature-branch` is outside of `daeuniverse`. For now, it is hard to fix. As a temporary solution, @MarksonHon please push your local branch branch to `origin daeuniverse` followed by the commands below:

```bash
git remote add upstream https://github.com/daeuniverse/daed
git fetch upstream --prune
git push -u upstream <your feature branch name>
```

**Comment by yqlbu** [comment]
> > @yqlbu instantiate-check-runs is causing PR build failure, please help
> 
> The root cause is that the `feature-branch` is outside of `daeuniverse`. For now, it is hard to fix. As a temporary solution, @MarksonHon please push your local branch branch to `origin daeuniverse` followed by the commands below:
> 
> ```shell
> git remote add upstream https://github.com/daeuniverse/daed
> git fetch upstream --prune
> git push -u upstream <your feature branch name>
> ```

Issue was resolved in #167

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -210,14 +239,25 @@ jobs:
...
-            cp bundled/daed-linux-$GOARCH$GOAMD64$GOARM.pacman installer-${{ steps.get_filename.outputs.BUNDLE_NAME }}.pkg.tar.zst
+            if [ "$ARCH_PACMAN" == 'pacman' ];then
+              cp bundled/daed-linux-$GOARCH$GOAMD64$GOARM.pacman installer-${{ steps.get_filename.outputs.BUNDLE_NAME }}.pkg.tar.zst
+            fi
+            ls | grep -E ".deb|.rpm|.pkg.tar.zst"
           fi
 
       - name: Upload Linux packages
-        if: ${{ env.GOARM != '5' && env.GOARM != '6' }}
+        if: ${{ env.GOARM != '5' && env.GOARM != '6' && env.GOARCH != 'mips' && env.GOARCH != 'mipsle' && env.GOARCH != 'mips64' && env.GOARCH != 'mips64le' }}
```
Comment (kunish) [inline .github/workflows/build.yml:249]:
```suggestion
        if: !contains(fromJSON('["5", "6"]'), env.GOARM) && !contains(fromJSON('["mips", "mipsle", "mips64", "mips64le"]'), env.GOARCH)
```

**Review by kunish — CHANGES_REQUESTED** [review]

**Comment by MarksonHon** [comment]
> I think we also need to update [pr-build.yml](https://github.com/daeuniverse/daed/blob/main/.github/workflows/pr-build.yml), [release.yml](https://github.com/daeuniverse/daed/blob/main/.github/workflows/release.yml), and [prerelease](https://github.com/daeuniverse/daed/blob/main/.github/workflows/prerelease.yml)

`pr-build.yml` has no package build flow, sync to `release.yml` and `prerelease.yml`

**Review by kunish — COMMENTED** [review]
It's too complicated for me to really have a good understanding what this PR is doing, can you leave a brief explanation to help people like me?

**Comment by MarksonHon** [comment]
> It's too complicated for me to really have a good understanding what this PR is doing, can you leave a brief explanation to help people like me?

https://github.com/daeuniverse/daed/pull/145#issuecomment-1636646309

Some packages built by Action cannot be installed on target operating system, such as Arch Linux ARM64, Debian armv7, this PR is coming for fixing this problem

**Review by kunish — APPROVED** [review]

**Pull request merged**